src/main/java/com/uber/cadence/internal/worker/PollerOptions.java

```diff
@@ -0,0 +1,157 @@
+package com.uber.cadence.internal.worker;
+
+import java.time.Duration;
+
+/**
+ * TODO: Switch to Duration.
+ */
+public final class PollerOptions {
+
+    public final static class Builder {
+
+        private int maximumPollRateIntervalMilliseconds = 1000;
+
+        private double maximumPollRatePerSecond;
+
+        private double pollBackoffCoefficient = 2;
+
+        private Duration pollBackoffInitialInterval = Duration.ofMillis(100);
+
+        private Duration pollBackoffMaximumInterval = Duration.ofMinutes(1);
+
+        private int pollThreadCount = 1;
+
+        private String pollThreadNamePrefix = "poller";
+
+        private Thread.UncaughtExceptionHandler uncaughtExceptionHandler;
+
+        public Builder() {
+        }
+
+        public Builder(PollerOptions o) {
+            if (o == null) {
+                return;
+            }
+            this.maximumPollRateIntervalMilliseconds = o.getMaximumPollRateIntervalMilliseconds();
+            this.maximumPollRatePerSecond = o.getMaximumPollRatePerSecond();
+            this.pollBackoffCoefficient = o.getPollBackoffCoefficient();
+            this.pollBackoffInitialInterval = o.getPollBackoffInitialInterval();
+            this.pollBackoffMaximumInterval = o.getPollBackoffMaximumInterval();
+            this.pollThreadCount = o.getPollThreadCount();
+            this.pollThreadNamePrefix = o.getPollThreadNamePrefix();
+            this.uncaughtExceptionHandler = o.getUncaughtExceptionHandler();
+        }
+
+        public Builder setMaximumPollRateIntervalMilliseconds(int maximumPollRateIntervalMilliseconds) {
+            this.maximumPollRateIntervalMilliseconds = maximumPollRateIntervalMilliseconds;
+            return this;
+        }
+
+        public Builder setMaximumPollRatePerSecond(double maximumPollRatePerSecond) {
+            this.maximumPollRatePerSecond = maximumPollRatePerSecond;
+            return this;
+        }
+
+        public Builder setPollBackoffCoefficient(double pollBackoffCoefficient) {
```

Comment:
its not clear to me as to whether these options are for

(1) Poller backoff/retry on errors ? (or)
(2) Poller throttling ?

Since this is exposed to the users through Builder, might be worth clarifying or simplifying this